pkg/controller/versions/util.go

```diff
@@ -65,41 +65,82 @@ func checkDiscoverySelectorsNotSet(spec *v2.ControlPlaneSpec, allErrors []error)
 	return allErrors
 }
 
-func validateGlobal(ctx context.Context, version Ver, meta *metav1.ObjectMeta, spec *v2.ControlPlaneSpec, cl client.Client, allErrors []error) []error {
-	isClusterScoped, err := version.Strategy().IsClusterScoped(spec)
-	if err != nil {
-		return append(allErrors, err)
-	}
+func validateGlobal(ctx context.Context, meta *metav1.ObjectMeta, newSmcp *v2.ControlPlaneSpec, cl client.Client, allErrors []error) []error {
 	smcps := v2.ServiceMeshControlPlaneList{}
-	err = cl.List(ctx, &smcps)
+	err := cl.List(ctx, &smcps)
 	if err != nil {
 		return append(allErrors, err)
 	}
 
-	if isClusterScoped {
-		// allow SMCP create/update only when no SMCP exists or when a single SMCP exists and we're updating it
-		if len(smcps.Items) > 1 || len(smcps.Items) == 1 && smcps.Items[0].UID != meta.GetUID() {
-			return append(allErrors,
-				fmt.Errorf("a cluster-scoped SMCP may only be created when no other SMCPs exist"))
+	otherSmcpExists := fmt.Errorf("a cluster-scoped SMCP may only be created when no other SMCPs exist")
+	if newSmcp.IsClusterScoped() {
+		// an SMCP already exists and new one is created
+		if len(smcps.Items) == 1 && smcps.Items[0].UID != meta.GetUID() {
+			currentSmcp := smcps.Items[0].Spec
+			if currentSmcp.IsClusterScoped() {
+				// allow cluster-wide gateway controller when another cluster-wide non gateway controller already exists
+				// this is the case where openshift-ingress controller and cluster-wide mesh co-exist
+				if (newSmcp.IsGatewayController() && !currentSmcp.IsGatewayController()) || (!newSmcp.IsGatewayController() && currentSmcp.IsGatewayController()) {
+					return append(allErrors, validateOverlappingCaCertConfigMapNames(meta, newSmcp, &smcps, allErrors)...)
+				}
+				// do not allow more than 1 cluster-wide gateway controller
+				if newSmcp.IsGatewayController() && currentSmcp.IsGatewayController() {
+					return append(allErrors, otherSmcpExists)
+				}
+			}
+			// only cluster-wide gateway controller can be created when another SMCP exist
+			if !newSmcp.IsGatewayController() {
+				return append(allErrors, otherSmcpExists)
+			}
+		}
+		if len(smcps.Items) > 1 {
+			if newSmcp.IsGatewayController() && countGatewayControllers(smcps.Items) > 1 ||
```

Comment:
You're right. I added a test case, which cannot exist and then I implemented conditions, and I didn't notice that it does not make any sense.
```
{
        name: "creating cluster-wide gateway controller SMCP when another already exists - expected error (2nd execution)",
        smcp: clusterWideGatewayController,
        existingObjs: []*maistrav2.ServiceMeshControlPlane{
                NewV2SMCPResource("basic", "istio-system-1", clusterWideGatewayController),
                NewV2SMCPResource("basic", "istio-system-2", clusterWideGatewayController2),
        },
        expectedErr: fmt.Errorf("a cluster-scoped SMCP may only be created when no other SMCPs exist"),
},
```

Reply:
I was wrong. We need that test case as well, because 2 gateway-controllers can exist when a user creates: gateway-controller and simple cluster-wide and then changes the 2nd to gateway-controller. That's why more than 1 gateway-controller can be found and in such a case we have to return an error and stop reconciliation.